provisioning/service/src/HttpDeviceRegistrationClient.cs

```diff
@@ -0,0 +1,660 @@
+                }
+            };
+
+            return this.httpClientHelper.DeleteAsync<IETagHolder>(GetDeviceRegistrationUri(registrationId), eTagHolder, errorMappingOverrides, null, cancellationToken);
+        }
+
+        private static Uri GetEnrollmentUri(string registrationId)
+        {
+            registrationId = WebUtility.UrlEncode(registrationId);
+            return new Uri(EnrollmentUriFormat.FormatInvariant(registrationId, ApiVersionQueryString), UriKind.Relative);
+        }
+
+        public static Uri GetBulkEnrollmentUri()
+        {
+            return new Uri(BulkEnrollmentUriFormat.FormatInvariant(ApiVersionQueryString), UriKind.Relative);
+        }
+
+        private static Uri GetQueryEnrollmentUri()
+        {
+            return new Uri(QueryEnrollmentUriFormat.FormatInvariant(ApiVersionQueryString), UriKind.Relative);
+        }
+
+        private static Uri GetQueryEnrollmentGroupUri()
+        {
+            return new Uri(QueryEnrollmentGroupUriFormat.FormatInvariant(ApiVersionQueryString), UriKind.Relative);
+        }
+
+        private static Uri GetEnrollmentGroupUri(string enrollmentGroupId)
+        {
+            enrollmentGroupId = WebUtility.UrlEncode(enrollmentGroupId);
+            return new Uri(EnrollmentGroupUriFormat.FormatInvariant(enrollmentGroupId, ApiVersionQueryString), UriKind.Relative);
+        }
+
+        private static Uri GetDeviceRegistrationUri(string registrationId)
+        {
+            registrationId = WebUtility.UrlEncode(registrationId);
+            return new Uri(DeviceRegistrationUriFormat.FormatInvariant(registrationId, ApiVersionQueryString), UriKind.Relative);
+        }
+
+        private static Uri GetDeviceRegistrationsUri(string enrollmentGroupId)
+        {
+            enrollmentGroupId = WebUtility.UrlEncode(enrollmentGroupId);
+            return new Uri(DeviceRegistrationsUriFormat.FormatInvariant(enrollmentGroupId, ApiVersionQueryString), UriKind.Relative);
+        }
+
+        static void ValidateRegistrationAndDeviceId(Enrollment enrollment)
+        {
+            if (enrollment == null)
+            {
+                throw new ArgumentNullException(nameof(enrollment));
+            }
+
+            if (string.IsNullOrWhiteSpace(enrollment.RegistrationId))
+            {
+                throw new ArgumentException("enrollment.RegistrationId");
+            }
+
+            if (!DeviceIdRegex.IsMatch(enrollment.RegistrationId))
+            {
+                throw new ArgumentException(ApiResources.EnrollmentIdNotValid.FormatInvariant(enrollment.RegistrationId));
+            }
+
+            if(!string.IsNullOrWhiteSpace(enrollment.DeviceId) &&
+               !DeviceIdRegex.IsMatch(enrollment.DeviceId))
+            {
+                throw new ArgumentException(ApiResources.EnrollmentIdNotValid.FormatInvariant(enrollment.DeviceId));
+            }
+        }
+
+        static void Validate(EnrollmentGroup enrollmentGroup)
```

Comment:
missing `internal`
Also the name is ambiguous. `ValidateEnrollmentGroup` maybe?